Mathlib/Logic/Equiv/Defs.lean

```diff
@@ -291,6 +291,11 @@ theorem eq_symm_apply {α β} (e : α ≃ β) {x y} : y = e.symm x ↔ e y = x :
 theorem symm_bijective : Function.Bijective (Equiv.symm : (α ≃ β) → β ≃ α) :=
   Function.bijective_iff_has_inverse.mpr ⟨_, symm_symm, symm_symm⟩
 
+@[simp]
+theorem symm_eq_self_of_involutive (f : Equiv.Perm α) (h : Function.Involutive f) : f.symm = f := by
```

Comment:
Could you add this related result:
```lean
theorem Function.Involutive.leftInverse_iff {f g : α → α} (h : Function.Involutive f) :
    Function.LeftInverse g f ↔ g = f :=
  ⟨fun hg => funext fun x => by rw [← h x, hg, h], fun he => he ▸ h.leftInverse⟩
```
which can then be used to prove your result as `h.leftInverse_iff.2 f.left_inv` or similar.

Reply:
I tried to make `h.leftInverse_iff.2 f.left_inv` work, but I couldn't improve on the version provided in the last commit. In particular, I couldn't write it in term mode (even when using `funext`).